src/options.rs

```diff
@@ -133,6 +133,10 @@ pub struct ResolveOptions {
     ///
     /// Default `false`
     pub builtin_modules: bool,
+
+    /// When enabled, will attempt to reduce memory usage by clearing data and fields
+    /// that are not relevant to the current process, and shuold not be held in memory.
+    pub reduce_memory_usage: bool,
```

Comment:
nit: I would call this option `remove_unused_fields` or `empty_unused_fields`. The term "memory usage" seems to be too broad and doesn't explain what it means. In fact, the proposed names get closer to the documentation.